commandapi-platforms/commandapi-velocity/commandapi-velocity-plugin/src/main/java/dev/jorel/commandapi/config/VelocityConfigurationAdapter.java

```diff
@@ -0,0 +1,105 @@
+package dev.jorel.commandapi.config;
+
+import org.spongepowered.configurate.CommentedConfigurationNode;
+import org.spongepowered.configurate.ConfigurationNode;
+import org.spongepowered.configurate.serialize.SerializationException;
+import org.spongepowered.configurate.yaml.YamlConfigurationLoader;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+@SuppressWarnings("ConfusingArgumentToVarargsMethod")
+public record VelocityConfigurationAdapter(YamlConfigurationLoader loader, CommentedConfigurationNode config) implements ConfigurationAdapter<ConfigurationNode, DefaultedVelocityConfig> {
+
+	@Override
+	public void setValue(String key, Object value) {
+		setValue0(key, value);
+	}
+
+	@Override
+	public void setComment(String key, String[] comment) {
+		config.node(key.split("\\.")).comment(String.join("\n", comment));
+	}
+
+	@Override
+	public Object getValue(String key) {
+		return getValue0(key);
+	}
+
+	@SuppressWarnings("DataFlowIssue")
+	@Override
+	public String[] getComment(String key) {
+		return config.node(key.split("\\.")).comment().split("\n");
+	}
```

Comment:
It doesn't seem like this `DataFlowIssue` should be suppressed. If a node doesn't have a comment on it, then `.comment()` will return `null` and `.split("\n")` will throw an exception.

To reproduce, put the plugin on a Velocity server to get this config file (for some reason, the first generation doesn't have any comments, which seems like a separate issue).

![image](https://github.com/user-attachments/assets/9e3e9d44-7387-4a5b-a951-1d17d67f01eb)

If you then stop and restart the server, I got this error:

```
[17:23:29 ERROR]: Can't create plugin commandapi
com.google.inject.ProvisionException: Unable to provision, see the following errors:

1) [Guice/ErrorInjectingConstructor]: NullPointerException: Cannot invoke "String.split(String)" because the return value of "org.spongepowered.configurate.CommentedConfigurationNode.comment()" is null
  at CommandAPIMain.<init>(CommandAPIMain.java:39)
  at VelocityPluginModule.configure(VelocityPluginModule.java:47)
  while locating CommandAPIMain

Learn more:
  https://github.com/google/guice/wiki/ERROR_INJECTING_CONSTRUCTOR

1 error

======================
Full classname legend:
======================
CommandAPIMain:       "dev.jorel.commandapi.CommandAPIMain"
VelocityPluginModule: "com.velocitypowered.proxy.plugin.loader.java.VelocityPluginModule"
========================
End of classname legend:
========================

        at com.google.inject.internal.InternalProvisionException.toProvisionException(InternalProvisionException.java:251) ~[server.jar:3.3.0-SNAPSHOT (git-6100e675-b416)]
        at com.google.inject.internal.InjectorImpl$1.get(InjectorImpl.java:1151) ~[server.jar:3.3.0-SNAPSHOT (git-6100e675-b416)]
        at com.google.inject.internal.InjectorImpl.getInstance(InjectorImpl.java:1186) ~[server.jar:3.3.0-SNAPSHOT (git-6100e675-b416)]
        at com.velocitypowered.proxy.plugin.loader.java.JavaPluginLoader.createPlugin(JavaPluginLoader.java:131) ~[server.jar:3.3.0-SNAPSHOT (git-6100e675-b416)]
        at com.velocitypowered.proxy.plugin.VelocityPluginManager.loadPlugins(VelocityPluginManager.java:168) ~[server.jar:3.3.0-SNAPSHOT (git-6100e675-b416)]
        at com.velocitypowered.proxy.VelocityServer.loadPlugins(VelocityServer.java:362) ~[server.jar:3.3.0-SNAPSHOT (git-6100e675-b416)]
        at com.velocitypowered.proxy.VelocityServer.start(VelocityServer.java:239) ~[server.jar:3.3.0-SNAPSHOT (git-6100e675-b416)]
        at com.velocitypowered.proxy.Velocity.main(Velocity.java:71) ~[server.jar:3.3.0-SNAPSHOT (git-6100e675-b416)]
Caused by: java.lang.NullPointerException: Cannot invoke "String.split(String)" because the return value of "org.spongepowered.configurate.CommentedConfigurationNode.comment()" is null
        at dev.jorel.commandapi.config.VelocityConfigurationAdapter.getComment(VelocityConfigurationAdapter.java:35) ~[?:?]
        at dev.jorel.commandapi.config.ConfigGenerator.generateWithNewValues(ConfigGenerator.java:52) ~[?:?]
        at dev.jorel.commandapi.CommandAPIMain.<init>(CommandAPIMain.java:67) ~[?:?]
        at dev.jorel.commandapi.CommandAPIMain$$FastClassByGuice$$a2b4c.GUICE$TRAMPOLINE(<generated>) ~[?:?]
        at dev.jorel.commandapi.CommandAPIMain$$FastClassByGuice$$a2b4c.apply(<generated>) ~[?:?]
        at com.google.inject.internal.DefaultConstructionProxyFactory$FastClassProxy.newInstance(DefaultConstructionProxyFactory.java:82) ~[server.jar:3.3.0-SNAPSHOT (git-6100e675-b416)]
        at com.google.inject.internal.ConstructorInjector.provision(ConstructorInjector.java:114) ~[server.jar:3.3.0-SNAPSHOT (git-6100e675-b416)]
        at com.google.inject.internal.ConstructorInjector.construct(ConstructorInjector.java:91) ~[server.jar:3.3.0-SNAPSHOT (git-6100e675-b416)]
        at com.google.inject.internal.ConstructorBindingImpl$Factory.get(ConstructorBindingImpl.java:300) ~[server.jar:3.3.0-SNAPSHOT (git-6100e675-b416)]
        at com.google.inject.internal.ProviderToInternalFactoryAdapter.get(ProviderToInternalFactoryAdapter.java:40) ~[server.jar:3.3.0-SNAPSHOT (git-6100e675-b416)]
        at com.google.inject.internal.SingletonScope$1.get(SingletonScope.java:169) ~[server.jar:3.3.0-SNAPSHOT (git-6100e675-b416)]
        at com.google.inject.internal.InternalFactoryToProviderAdapter.get(InternalFactoryToProviderAdapter.java:45) ~[server.jar:3.3.0-SNAPSHOT (git-6100e675-b416)]
        at com.google.inject.internal.InjectorImpl$1.get(InjectorImpl.java:1148) ~[server.jar:3.3.0-SNAPSHOT (git-6100e675-b416)]
        ... 6 more
```

Reply:
> for some reason, the first generation doesn't have any comments, which seems like a separate issue

The first generation? Does restarting the server (or rather the proxy) create the comments? Because I've brought this up on discord before, apparently Configurate's Yaml doesn't support comments and other formats might not be better for us.

Reply:
Well, the first time there are no comments as pictured. The second time, I get that error and nothing updates because the code is interrupted. Sooo... I can't tell if restarting the server would add comments because the plugin crashes :P.

Reply:
Hmmm.. I wonder if we could just not use the CommentedConfigurationNode since comments are not really a thing here. Ideally we get support for that but for now I am thinking to leave out comments on Velocity.
So yeah, it obviously crashes when comments aren't present (which should be avoided).